src/urbit/app/chess.hoon

```diff
@@ -778,56 +886,96 @@
 ::  helper core for moves
 ::  test if a given move is legal
 ++  try-move
-  |=  [game=chess-game position=chess-position move=chess-move]
-  ^-  [new=(unit [game=chess-game position=chess-position]) cards=(list card)]
-  ?.  ?=(~ result.game)
+  |=  [game=active-game-state move=chess-move]
+  ^-  [new=(unit active-game-state) cards=(list card)]
+  ?.  ?=(~ result.game.game)
     [~ ~]
   =/  new-position
-    (~(apply-move with-position position) move)
+    (~(apply-move with-position position.game) move)
   ?~  new-position
     [~ ~]
-  =/  updated-game  `chess-game`game
+  =/  updated-game  `chess-game`game.game
   =.  moves.updated-game  (snoc moves.updated-game move)
+  =/  new-fen-repetition  (increment-repetition fen-repetition.game u.new-position)
   =/  fen  (position-to-fen u.new-position)
   =/  position-update-card
-    :*  %give  %fact  ~[/game/(scot %da game-id.game)/updates]
-        %chess-update  !>([%position game-id.game fen])
+    :*  %give  %fact  ~[/game/(scot %da game-id.game.game)/updates]
+        %chess-update  !>([%position game-id.game.game fen])
+    ==
+  =/  move-card
+    :*  %give  %fact  ~[/game/(scot %da game-id.game.game)/moves]
+                      %chess-move  !>(move)
     ==
   =/  in-checkmate  ~(in-checkmate with-position u.new-position)
   =/  in-stalemate  ?:  in-checkmate
                       |
                     ~(in-stalemate with-position u.new-position)
   ::  check if game ends by checkmate or stalemate
-  ?:  ?|  ?=(%end -.move)
-          in-checkmate
+  =/  threefold-draw  &((check-threefold new-fen-repetition u.new-position) auto-claim-special-draws.game)
+  ?:  ?|  in-checkmate
           in-stalemate
+          threefold-draw
       ==
       ::  update result with score
       =.  result.updated-game
-        ?:  ?=(%end -.move)  `+.move
         ?:  in-stalemate  `%'½–½'
+        ?:  threefold-draw  `%'½–½'
         ?:  in-checkmate
           ?-  player-to-move.u.new-position
             %white  `%'0-1'
             %black  `%'1-0'
           ==
         !!
       ::  give a card of the game result to opponent ship
-      :-  `[updated-game u.new-position]
+      :-  `[updated-game u.new-position new-fen-repetition |3.game]
+      ?.  threefold-draw
+        :~  position-update-card
+            move-card
```

Comment:
I think there's a bug here: I think we're emitting `move-card` regardless of whether we're verifying our own move or the opponents